### PR TITLE
Document Edit Command Implementation and Testing

### LINKED ISSUE #260: Document Edit Command in UG and DG
The `edit` command lacks comprehensive documentation in both the User Guide and Developer Guide. This task involves updating the command's usage rules, technical implementation details, and manual testing instructions.

### Proposed Changes
* **User Guide**: Detail the `edit` command format, duplicate detection rules (name + date overlap), and tag clearing behavior.
* **Developer Guide**: Document the `EditTripDescriptor` mechanism and logic flow for partial updates.
* **Manual Testing**: Add test cases to the DG appendix covering valid/invalid date ranges and duplicate prevention during edits.

### Tasks
- [ ] Update `edit` section in `docs/UserGuide.md`.
- [ ] Add `edit` implementation details to `docs/DeveloperGuide.md`.
- [ ] Append `edit` test cases to the manual testing section in the DG.
- [ ] Verify documentation accuracy against `EditCommand.java` and `EditCommandParser.java`.

### PR DESCRIPTION
**Pull request by ths1959**
This PR resolves #260 by providing comprehensive documentation for the `edit` command in both the User Guide (UG) and Developer Guide (DG). It clarifies partial update logic, duplicate detection rules based on date overlaps, and adds manual testing instructions.

### Key Implementation Details
* **Edit Logic**: Documented the use of `EditTripDescriptor` as a temporary container for partial updates.
* **Validation**: Explained the integration of `hasTripExcluding` to prevent duplicate creation while allowing name changes.
* **Date Constraints**: Detailed the enforcement of chronological validity (start ≤ end) during field-specific edits.

### Documentation
* **User Guide**: Updated the `edit` section with tag-clearing behavior and specific date-range rules.
* **Developer Guide**: Added a new implementation subsection for the `edit` command and expanded the manual testing appendix with seven distinct edge-case scenarios.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -442,6 +456,37 @@ testers are expected to do more *exploratory* testing.
     1. Test case: `add sd/2026-06-01 ed/2026-06-10`
        Expected: Error message indicating invalid command format. No trip is added.
 
+### Editing a trip
```
Comment (derrikzzz) [inline docs/DeveloperGuide.md]: Perhaps can include a "no change" check here, referenced from `EditCommand.java`
```
if (tripToEdit.equals(editedTrip)) {
    throw new CommandException(MESSAGE_NO_CHANGES);
}
```

**Review by derrikzzz — COMMENTED** [review]
Changes looks good to ship!  🚀
